website/docs/r/vapp_vm.html.markdown

```diff
@@ -97,13 +116,29 @@ The following arguments are supported:
 * `cpu_cores` - (Optional; *v2.1+*) The number of cores per socket
 * `metadata` - (Optional; *v2.2+*) Key value map of metadata to assign to this VM
 * `initscript` (Optional) A script to be run only on initial boot
-* `network_name` - (Optional) Name of the network this VM should connect to
-* `vapp_network_name` - (Optional; *v2.1+*) Name of the vApp network this VM should connect to
-* `ip` - (Optional) The IP to assign to this vApp. Must be an IP address or
-  one of "dhcp", "allocated" or "none". If given the address must be within the
+* `network_name` - (**Deprecated**) Name of the network this VM should connect to. **Conflicts** with `networks`.
+* `vapp_network_name` - (**Deprecated**; *v2.1+*) Name of the vApp network this VM should connect to. **Conflicts** with `networks`.
```

Comment:
This makes me wonder whether we shouldn't add `ConflictsWith` in the code itself to protect the user from these mistakes. Unfortunately, it doesn't seem this keyword is widely documented...
https://www.terraform.io/docs/extend/schemas/schema-methods.html
https://www.terraform.io/docs/extend/best-practices/deprecations.html

Reply:
It is set. Resources have `ConflictsWith:    []string{"networks"},` while `networks` has `ConflictsWith: []string{"ip", "network_name", "vapp_network_name"},`